core/src/HIP/Kokkos_HIP_Team.hpp

```diff
@@ -79,8 +81,14 @@ class HIPTeamMember {
 
  public:
   KOKKOS_INLINE_FUNCTION
-  const execution_space::scratch_memory_space& team_shmem() const {
-    return m_team_shared.set_team_thread_mode(0, 1, 0);
+  const auto& team_shmem() const {
```

Comment:
```suggestion
  const scratch_memory_space_l0& team_shmem() const {
```

is there a reason why this deviates from the other backends?

Reply:
No, not really. The real question is if this should return the generic space instead (if we decide that we can assign from the specialized spaces to the generic one) or if changing the type is fine.